### ISSUE TITLE
Little bugs and Feature Request: share mentor discussions?

### ISSUE THREAD

**Issue by vladic4t** [exercism/exercism#5823]
Hello,

I would like to report the following two bugs:

1. After the exercismv3 upgrade comments that were left by the users on their solutions and were not reviewed before 1 september by a mentor have all disappeared (at least for me).

2. There is no "quote" button in the comment box: [https://imgur.com/a/5AHkfO8](url)

Feature request:

I feel like taking part in mentoring has become more friendly and more people is starting to join. Maybe it might be a good idea for mentoring discussions to be rated on the mentor side to allow all mentors to learn from them. Taking privacy into consideration of course and ask the user what information he wants to disclose and/or even asking the user if he wants the discussion to be open at all. And maybe also have an extra reputation for that along the way. All win :)

**Comment by iHiD** [comment]
> After the exercismv3 upgrade comments that were left by the users on their solutions and were not reviewed before 1 september by a mentor have all disappeared (at least for me).

Could you give me an example of this pls?

**Comment by iHiD** [comment]
> There is no "quote" button in the comment box: https://imgur.com/a/5AHkfO8

Could you open a separate issue for this pls?

**Comment by iHiD** [comment]
> I feel like taking part in mentoring has become more friendly and more people is starting to join. Maybe it might be a good idea for mentoring discussions to be rated on the mentor side to allow all mentors to learn from them. Taking privacy into consideration of course and ask the user what information he wants to disclose and/or even asking the user if he wants the discussion to be open at all. And maybe also have an extra reputation for that along the way. All win :)

Mentors will be able to share solutions with each other and everything will be anonymised. Coming at some point in the next month.

**Comment by vladic4t** [comment]
> Could you give me an example of this pls?

Yes. You can find my account at exercism https://exercism.org/profiles/vladic4t.

For example, I submitted various iterations for the exercise _Sieve_ in the track C++ **21 days ago** from today (i.e. before 1 September) and each iteration contained a comment (some more like 2 comments) written by me.

After the upgrade, all those comments are no longer there.
The link (v3) for the discussion is https://exercism.org/tracks/cpp/exercises/sieve/mentor_discussions/2ee98079fd5848d89cf06e3222cd7acd

I'm aware the given link can not be accessed from any user; it yields a 404. I assume, however, with higher privileges the content can be accessed. Let me know if you need anything else.

For me particularly this issue is not really a big deal, even though I had large comments written. For others it might affect the mentor experience. I don't know.

**Comment by vladic4t** [comment]
> > I feel like taking part in mentoring has become more friendly and more people is starting to join. Maybe it might be a good idea for mentoring discussions to be rated on the mentor side to allow all mentors to learn from them. Taking privacy into consideration of course and ask the user what information he wants to disclose and/or even asking the user if he wants the discussion to be open at all. And maybe also have an extra reputation for that along the way. All win :)
> 
> Mentors will be able to share solutions with each other and everything will be anonymised. Coming at some point in the next month.

Sounds cool! How _everything will be anonymised_ is a problem engineers need to think about carefully. I can't really say much because I'm not an expert myself and I have not spent days thinking about it. However, in my opinion, I would like to always remain flexible enough on what I can talk about with the mentor as a learner including my thought process while leaving the relevant discussion of the exercise mentored public. That's why I thought allowing a party what information to disclose might allow better discussions with the mentor while also not sacrificing the quality of the mentoring. This is from my perspective, maybe it's something not to worry about!

**Comment by vladic4t** [comment]
Also important. I don’t know what consequences this might bring but it’s worth thinking. I, as multicultural user, have already found myself using three different languages (not programming ones lol) during mentor sessions as I sometimes prefer communicating with one’s native language.

I don’t how you might approach this. Maybe either by forcing the use of the English language (a very simple solution indeed) on both sides or you might as well consider implementing an api for automatic translation when needed (google translate or deepl) for the mentor. Now I feel like this overlaps to something exercism lacks entirely which is the internationalization of the site. But maybe this is a first step? ;)

**Comment by SleeplessByte** [comment]
It will be anonymised like in v2, where it redacts the usernames. I recommend @iHiD to also redact the name's as many mentors will do something like Hi first_name (instead of Hi username).

**Comment by szabgab** [comment]
Will the mentoring session be totally public? If not totally public will students be able to share a mentoring session by someone else to look at? e.g. by sending a link or by adding their user-id to the "list of approved viewers" and then maybe sending a link.
